akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2Multiplexer.scala

```diff
@@ -43,45 +45,57 @@ private[http2] trait Http2MultiplexerSupport { logic: GraphStageLogic with Stage
     new Http2Multiplexer with OutHandler with StateTimingSupport with LogSupport {
       outlet.setHandler(this)
 
-      class OutStream(
+      class OutStream[T](
         val streamId:           Int,
-        private var maybeInlet: Option[SubSinkInlet[ByteString]],
+        private var maybeInlet: Option[SubSinkInlet[T]],
```

Comment:
I think the extra type parameter can be removed if you use `SubSinkInlet[_]` here. We match on the runtime time of the elements anyway below.

Reply:
Yeah might be nice to try out both variants (I somewhat avoided the runtime match in #1858)